### PR TITLE
convert ph2.c to D

### PR DESCRIPTION
**Pull request by WalterBright**
The deleted section was only for dmc, and was not used by dmc.

### CODE REVIEW

**Comment by thewilsonator** [comment]
Semaphore timeout.

File: src/dmd/backend/ph2.d [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+    {
+        hprev = h.prev;
+        free(h.buf);
+        free(h);
+    }
+}
+}
+
+void ph_newheap(size_t nbytes)
+{   uint newsize;
```
Comment (RazvanN7) [inline src/dmd/backend/ph2.d]: newline

File: src/dmd/backend/ph2.d [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+        err_nomem();
+    }
+    h.nleft = newsize;
+    h.p = h.buf;
+    h.prev = heap;
+    heap = h;
+}
+
+void *ph_malloc(size_t nbytes)
+{   ubyte *p;
```
Comment (RazvanN7) [inline src/dmd/backend/ph2.d]: newline

File: src/dmd/backend/ph2.d [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+    p = heap.p;
+    heap.p += nbytes;
+    heap.nleft -= nbytes;
+    *cast(uint *)p = cast(uint)(nbytes - uint.sizeof);
+    p += uint.sizeof;
+    return p;
+}
+
+void *ph_calloc(size_t nbytes)
+{   void *p;
```
Comment (RazvanN7) [inline src/dmd/backend/ph2.d]: newline

File: src/dmd/backend/ph2.d [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+
+    nbytes += uint.sizeof * 2;
+    nbytes &= ~(uint.sizeof - 1);
+
+    if (nbytes >= heap.nleft)
+        ph_newheap(nbytes);
+    p = heap.p;
+    heap.p += nbytes;
+    heap.nleft -= nbytes;
+    *cast(uint *)p = cast(uint)(nbytes - uint.sizeof);
```
Comment (RazvanN7) [inline src/dmd/backend/ph2.d]: no space between uint and *

**Review by RazvanN7 — APPROVED** [review]
LGTM minus the style comments

File: src/dmd/backend/ph2.d (hunk 1 above)
Comment (jacob-carlborg) [reply src/dmd/backend/ph2.d]: Walter usually waits with the style changes to after the file is converted.

**Pull request merged**